functests/utils/mcps/mcps.go

```diff
@@ -0,0 +1,129 @@
+package mcps
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/klog"
+	"time"
+
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	machineconfigv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+
+	testclient "github.com/openshift-kni/performance-addon-operators/functests/utils/client"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/nodes"
+	"github.com/openshift-kni/performance-addon-operators/pkg/controller/performanceprofile/components"
+)
+
+const (
+	mcpUpdateTimeoutPerNode = 20
+)
+
+// GetByLabel returns all MCPs with the specified label
+func GetByLabel(key, value string) ([]machineconfigv1.MachineConfigPool, error) {
+	selector := labels.NewSelector()
+	req, err := labels.NewRequirement(key, selection.Equals, []string{value})
+	if err != nil {
+		return nil, err
+	}
+	selector = selector.Add(*req)
+	mcps := &machineconfigv1.MachineConfigPoolList{}
+	if err := testclient.Client.List(context.TODO(), mcps, &client.ListOptions{LabelSelector: selector}); err != nil {
+		return nil, err
+	}
+	return mcps.Items, nil
+}
+
+// GetByName returns the MCP with the specified name
+func GetByName(name string) (*machineconfigv1.MachineConfigPool, error) {
+	mcp := &machineconfigv1.MachineConfigPool{}
+	key := types.NamespacedName{
+		Name:      name,
+		Namespace: "",
```

Comment:
Just a nit, but no need to specify Namespace if it's empty string.

Reply:
Fair point, but actually I like stating explicitly that no namespace is needed. But there is a nicer way to do that, will change it to `metav1.NamespaceNone` (which is `""`, but nicer), which we use at other places as well.